src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitRouterIntegrationTest.kt

```diff
@@ -0,0 +1,193 @@
+
+                        connectAndCheck(rabbitMQContainer, queue, listOf(
+                            Expectation(messageA, 1, false, ManualAckDeliveryCallback.Confirmation::confirm),
+                            Expectation(messageB, 2, false, ManualAckDeliveryCallback.Confirmation::reject),
+                            Expectation(messageC, 3, false) { },
+                            Expectation(messageD, 4, false) { },
+                        ))
+
+                        connectAndCheck(rabbitMQContainer, queue, listOf(
+                            Expectation(messageC, 1, true, ManualAckDeliveryCallback.Confirmation::confirm),
+                            Expectation(messageD, 2, true) { },
+                        ))
+
+                        connectAndCheck(rabbitMQContainer, queue, listOf(
+                            Expectation(messageD, 1, true, ManualAckDeliveryCallback.Confirmation::reject),
+                        ))
+
+                        connectAndCheck(rabbitMQContainer, queue, emptyList())
+                    }
+                }
+            }
+    }
+
+    private fun connectAndCheck(
+        rabbitMQContainer: RabbitMQContainer,
+        queue: ArrayBlockingQueue<Delivery>,
+        expectations: List<Expectation>,
+    ) {
+        createConnectionManager(rabbitMQContainer).use { manager ->
+            createRouter(manager).use { router ->
+                val monitor = router.subscribeWithManualAck({ deliveryMetadata, message, confirmation ->
+                    queue.put(Delivery(message, deliveryMetadata.deliveryTag, deliveryMetadata.isRedelivered, confirmation))
+                })
+
+                try {
+                    expectations.forEach { expectation ->
+                        val delivery = assertNotNull(queue.poll(1, TimeUnit.SECONDS))
+                        assertEquals(expectation.message, delivery.message, "Message")
+                        assertEquals(expectation.deliveryTag, delivery.deliveryTag, "Delivery tag")
+                        assertEquals(expectation.redelivery, delivery.redelivery, "Redelivery flag")
+                        expectation.action.invoke(delivery.confirmation)
+                    }
+
+                    assertNull(queue.poll(1, TimeUnit.SECONDS))
+                } finally {
+                    monitor.unsubscribe()
+                }
+            }
+
+            createRouter(manager).use { router ->
+                val monitor = router.subscribeWithManualAck({ deliveryMetadata, message, confirmation ->
+                    queue.put(Delivery(message, deliveryMetadata.deliveryTag, deliveryMetadata.isRedelivered, confirmation))
+                })
+
+                try {
+                    // RabbitMQ doesn't resend messages after resubscribe using the same connection and channel
+                    assertNull(queue.poll(1, TimeUnit.SECONDS))
+                } finally {
+                    monitor.unsubscribe()
+                }
```

Comment:
Do we need to test rabbitmq here?.. This part does exactly this in my opinion.
We just need to make sure that after resubscribing we receive the expected messages from the queue, I think

Reply:
I spend some times to understand that rabbitmq doesn't resend message when we use the same connection and channel. I would like to illustrate this case in the test

Reply:
Then maybe we should add this information to javadoc for subscribe methods, what do you think? Because I don't believe that anyone who uses the common lib will go to its tests to check how the method behaves in this particular case